### PR TITLE
feat(native): Improvements for AppStoreConnect

### PR DESCRIPTION
**Pull request by mitsuhiko**
This includes some smaller fixes to make the processing work

### CODE REVIEW

File: src/sentry/models/debugfile.py [hunk 1]
```diff
@@ -422,7 +426,11 @@ def detect_dif_from_path(path, name=None, debug_id=None, accept_unknown=False):
             # Assume the basename is the debug_id, if it wasn't symbolic will fail.  This is
             # required for when we get called for files extracted from a zipfile.
             basename = os.path.basename(path)
-            debug_id = os.path.splitext(basename)[0]
+            try:
+                debug_id = normalize_debug_id(os.path.splitext(basename)[0])
+            except SymbolicError as e:
+                logger.debug("File does not look like a debug ID: %s", path)
+                raise BadDif("Invalid UuidMap: %s" % e)
```
Comment (Swatinem) [inline src/sentry/models/debugfile.py:433]: do we have to catch this in the appstore task?

**Review by Swatinem — APPROVED** [review]

File: src/sentry/models/debugfile.py (hunk 1 above)
Comment (mitsuhiko) [reply src/sentry/models/debugfile.py:433]: that already happens.

File: src/sentry/lang/native/appconnect.py [hunk 2]
```diff
@@ -383,4 +383,13 @@ def list_builds(self, kind: BuildKind = BuildKind.ALL) -> List[BuildInfo]:
...
                     )
                     builds.append(build)
+
+        def _try_int(x: str) -> int:
+            try:
+                return int(x)
+            except ValueError:
+                return 0
+
+        builds.sort(key=lambda x: _try_int(x.build_number), reverse=True)
```
Comment (flub) [inline src/sentry/lang/native/appconnect.py:393]: For future souls, please describe why this is sorted.  The API does treat these `build_number`s as opaque strings (it calls them `version`) so this is a bit of a stretch.

I think I'd prefer to make `BuildInfo` ordered instead and leave it up to the caller to sort this list.  That puts the sorting logic in the download task where it belongs.

File: src/sentry/models/debugfile.py [hunk 3]
```diff
@@ -422,7 +426,11 @@ def detect_dif_from_path(path, name=None, debug_id=None, accept_unknown=False):
             # Assume the basename is the debug_id, if it wasn't symbolic will fail.  This is
             # required for when we get called for files extracted from a zipfile.
             basename = os.path.basename(path)
-            debug_id = os.path.splitext(basename)[0]
+            try:
+                debug_id = normalize_debug_id(os.path.splitext(basename)[0])
+            except SymbolicError as e:
+                logger.debug("File does not look like a debug ID: %s", path)
```
Comment (flub) [inline src/sentry/models/debugfile.py:432]:
```suggestion
                logger.debug("Filename does not look like a debug ID: %s", path)
```

File: src/sentry/tasks/app_store_connect.py [hunk 4]
```diff
@@ -64,7 +64,12 @@ def inner_dsym_download(
 
         if not build_state.fetched:
             with tempfile.NamedTemporaryFile() as dsyms_zip:
-                itunes_client.download_dsyms(build, pathlib.Path(dsyms_zip.name))
+                try:
+                    itunes_client.download_dsyms(build, pathlib.Path(dsyms_zip.name))
+                except appconnect.NoDsymsError:
+                    # This build does not have dsyms
+                    logger.debug("No dSYMs for build %s", build)
```
Comment (flub) [inline src/sentry/tasks/app_store_connect.py:71]:
```suggestion
                    logger.debug("No dSYMs for build %s", build)
```

**Review by flub — APPROVED** [review]
I'm also happy to push my suggested changes to this PR if you like.

File: src/sentry/lang/native/appconnect.py (hunk 2 above)
Comment (jan-auer) [reply src/sentry/lang/native/appconnect.py:393]: You can also check whether there's a way to sort this via the API request. I've seen this in the ASC web app, although I didn't check if we're using the same API or not.

Comment (Swatinem) [reply src/sentry/lang/native/appconnect.py:395]: The `iris` (= web-facing API) has a `sort=-version` param for the `builds` endpoint. Though we are not using that endpoint here, so not sure if the same params are available for the officially documented API.

**Review by jan-auer — APPROVED** [review]
Let's get this merged, we can still improve this further in follow-up PRs.

**Pull request merged**